unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -1,144 +1,192 @@
 
-                if (amount >= CullingControllerSettings.MAX_POPULATING_ELEMENTS_PER_FRAME)
+                foreach (Renderer renderer in rendereable.renderers)
                 {
-                    yield return null;
-                    amount = 0;
-                }
-                amount++;
+                    if (renderer == null || !renderer.gameObject.activeInHierarchy)
+                        continue;
 
-                if ((((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
-                    && !(renderer is ParticleSystemRenderer))
-                {
-                    if (renderer is SkinnedMeshRenderer)
-                        skinnedRenderersList.Add((SkinnedMeshRenderer)renderer);
-                    else
-                        renderersList.Add(renderer);
+                    if (((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
+                        usingRenderers.Add(renderer);
                 }
             }
-            renderers = renderersList;
-            skinnedRenderers = skinnedRenderersList;
-
-            yield return null;
+        }
 
-            animations = Object.FindObjectsOfType<Animation>();
+        private void ForceCalculateSkinnedRenderers()
+        {
+            usingSkinnedRenderers = new List<SkinnedMeshRenderer>(detectedSkinnedRenderers);
+            foreach (SkinnedMeshRenderer skinnedRenderer in detectedSkinnedRenderers)
+            {
+                if (skinnedRenderer == null)
+                    continue;
 
-            dirty = false;
+                if (((1 << skinnedRenderer.gameObject.layer) & ignoredLayersMask) == 0)
+                    usingSkinnedRenderers.Add(skinnedRenderer);
+            }
         }
 
-        /// <summary>
-        ///  This will re-populate all the tracked objects in a sync way.
-        /// </summary>
-        /// <param name="includeInactives">True for add inactive objects to the tracked list.</param>
-        public void ForcePopulateRenderersList(bool includeInactives)
+        private void ForceCleanRendereables()
         {
-            List<Renderer> renderersList = new List<Renderer>();
-            List<SkinnedMeshRenderer> skinnedRenderersList = new List<SkinnedMeshRenderer>();
-            Renderer[] allRenderers = Object.FindObjectsOfType<Renderer>(includeInactives);
+            for (int i = rendereables.Count - 1; i >= 0; i--)
+            {
+                if (rendereables[i] == null || rendereables[i].container == null)
+                    rendereables.RemoveAt(i);
+            }
+        }
+
+        private bool ShouldNotBeIgnored(Renderer renderer) => ((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0;
+
 
-            foreach (Renderer renderer in allRenderers)
+        private IEnumerator CalculateRenderers()
+        {
+            float currentStartTime = Time.realtimeSinceStartup;
+            usingRenderers.Clear();
+            for (int i = rendereables.Count - 1; i >= 0; i--)
```

Comment:
is there a special reason to do it via reversed `for `loop and not `foreach`?

Reply:
Yes, we add the possibility to remove elements of the list as it goes